opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -313,12 +313,16 @@ namespace Opm
                     
                     const bool hasGroupControl = this->isInjector() ? inj_controls.hasControl(Well::InjectorCMode::GRUP) :
                                                                       prod_controls.hasControl(Well::ProducerCMode::GRUP);
-                    bool isGroupControl = ws.production_cmode == Well::ProducerCMode::GRUP || ws.injection_cmode == Well::InjectorCMode::GRUP; 
-                    if (! (isGroupControl && !this->param_.check_group_constraints_inner_well_iterations_)) {
-                        changed = this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
-                    }
-                    if (hasGroupControl && this->param_.check_group_constraints_inner_well_iterations_) {
-                        changed = changed || this->checkGroupConstraints(well_state, group_state, schedule, summary_state,deferred_logger);
+                    bool isGroupControl = ws.production_cmode == Well::ProducerCMode::GRUP || ws.injection_cmode == Well::InjectorCMode::GRUP;
+
+                    if (this->param_.check_group_constraints_inner_well_iterations_) {
+                        if (!isGroupControl && hasGroupControl) {
+                            changed = changed || this->checkGroupConstraints(well_state, group_state, schedule, summary_state,deferred_logger);
+                        }
+                        changed = changed || this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
+                    } else {
+                        if (!isGroupControl)
+                            changed = changed || this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
```

Comment:
Just for discussion, should not we always check individual constraints? Here, if `check_group_constraints_inner_well_iterations_` is false, and well is under group, then it will not do any constraint checking. It looks strange from the code level.

Reply:
In current master switching to and from GRUP during local switching is problematic since the reduction rate is not updated. I added "check_group_constraints_inner_well_iterations_" as a fast work-around. The goal of this PR is to fix this so we can switch to and from GRUP during local iterations. I plan to keep the parameter for some time, but if we don't see more issues, we should remove.

Reply:
Thanks for the explanation. 

I meant for the else branch (check_group_constraints_inner_well_iterations_ is false)
```c++
else {
                        if (!isGroupControl)
                            changed = changed || this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
```

if the well is under group control, this branch will not check any constraints, including the individual constraints.

Is this intentional?

Reply:
Yes. If check_group_constraints_inner_well_iterations_=false. We neither want to switch to GRUP or from GRUP. This is the same logic that we have in master.